doc/source.rst

```diff
@@ -181,11 +181,12 @@ File patterns
 File path patterns are used for include and omit, and for combining path
 remapping.  They follow common shell syntax:
 
-- ``*`` matches any number of file name characters, not including the directory
-  separator.
-
 - ``?`` matches a single file name character.
 
+- ``*`` matches any number of file name characters, not including the directory
+  separator.  As a special case, if a pattern starts with ``*/``, it is treated
+  as ``**/``, and if a pattern ends with ``/*``, it is treated as ``/**`.
```

Comment:
Having reviewed other places, I'm starting to get a feeling that things like `x/**z/y` and `s/x**z/y` won't work the way it's implied here either. How about adding more examples to this as well?